src/iqm/qiskit_iqm/iqm_backend.py

```diff
@@ -90,63 +104,52 @@ def __init__(
-        target = Target()
-
-        def _create_properties(
-            op_name: str, symmetric: bool = False
-        ) -> dict[tuple[int, ...], InstructionProperties | None]:
-            """Creates the Qiskit instruction properties dictionary for the given IQM native operation.
-
-            Currently we do not provide any actual properties for the operation other than the
-            allowed loci.
-            """
-            loci = operations[op_name]
-            if symmetric:
-                # For symmetric gates, construct all the valid loci for Qiskit.
-                # Coupling maps in Qiskit are directed graphs, and gate symmetry is provided explicitly.
-                loci = tuple(permuted_locus for locus in loci for permuted_locus in itertools.permutations(locus))
-            return {tuple(qb_to_idx[qb] for qb in locus): None for locus in loci}
+        # Because of the way the Target and the transpiler interact, the resonators need to have higher indices than
+        # qubits, or else transpiling with optimization_level=0 will fail because of lacking resonator indices.
+        qb_to_idx = {
+            qb: idx
+            for idx, qb in enumerate(
+                sorted(arch.qubits, key=_component_sort_key)
+                + sorted(arch.computational_resonators, key=_component_sort_key)
+            )
+        }
 
-        if 'measure' in operations:
-            target.add_instruction(Measure(), _create_properties('measure'))
-        target.add_instruction(
-            IGate(),
-            {(qb_to_idx[qb],): None for qb in arch.components},
-        )
-        if 'prx' in operations:
-            target.add_instruction(RGate(Parameter('theta'), Parameter('phi')), _create_properties('prx'))
-        if 'cz' in operations:
-            target.add_instruction(CZGate(), _create_properties('cz', symmetric=True))
-        if 'move' in operations:
-            target.add_instruction(MoveGate(), _create_properties('move'))
-        if 'cc_prx' in operations:
-            # HACK reset gate shares cc_prx loci for now
-            target.add_instruction(Reset(), _create_properties('cc_prx'))
-
-        self._target = target
+        self._target = IQMTarget(arch, qb_to_idx)
         self._qb_to_idx = qb_to_idx
         self._idx_to_qb = {v: k for k, v in qb_to_idx.items()}
         self.name = 'IQMBackend'
+        self._coupling_map = self.target.build_coupling_map()
 
     @property
     def target(self) -> Target:
         return self._target
 
-    def qubit_name_to_index(self, name: str) -> Optional[int]:
+    @property
+    def physical_qubits(self) -> list[str]:
+        """Return the list of physical qubits in the backend."""
+        return list(self._qb_to_idx)
+
+    def has_resonators(self) -> bool:
+        """Return whether the backend has resonators."""
+        return bool(self.architecture.computational_resonators)
```

Comment:
Is this needed? Can't the user just look at backend.architecture?

Reply:
They need to know that they can find it there. That's quite some archeology